### PR TITLE
RIX Single Point Operation

### PR DESCRIPTION
**Pull request by ghalym**
Implemented additional Fast Fault for the following SPO Ranges.
Implemented Veto logic based on the status of M1K1.
Increased the soft limits on M1K1 bender US and DS as requested during last comm shift.
Fixed random SP1K1 RTD linking pragma.

ToDo: IOC. Test and verify Zos encoder limit range.

SPO Ranges
image

Limits
image

### CODE REVIEW

**Comment by jyotiphy** [comment]
Maggie, implementation of new fast faults looks good! 
1. For ZOS height, should the sign be revered ?
2. Also, I'm still wondering if it makes more sense to have M1K1 bender fast faults coupled with these new fast faults, like we have these fast faults (together-with or) only when we are out of 'yellow' safe region for M1K1-Bender ? Isn't it overkill to  have these fast faults being in safe bender region, like tripping beam for no reason ?

**Review by jyotiphy — COMMENTED** [review]
Maggie, implementation of new fast faults looks good!

1. For ZOS height, should the sign be revered ?
2. Also, I'm still wondering if it makes more sense to have M1K1 bender fast faults coupled with these new fast faults, like we have these fast faults (together-with or) only when we are out of 'yellow' safe region for M1K1-Bender ? Isn't it overkill to have these fast faults being in safe bender region, like tripping beam for no reason ?

**Comment by ghalym** [comment]
1. Yeah, I need to test the ZOS. As I was doing the calculations, higher values in mm yields lower encoder counts.
2. Alex W. Wanted it this way. He wanted to keep the bender fast fault and bypass it when all these faults are green.

File: lcls-plc-rixs-optics/rixs_optics/POUs/PRG_SPO_PMPS.TcPOU [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	ffSP1K1GHorisontal : FB_FastFault := (
+        i_xAutoReset := TRUE,
+        i_DevName := 'SP1K1 Grating horizontal',
+        i_Desc := 'SP1K1 horizontal Pitch is out of Single Point Operation safe range',
+        i_TypeCode := 16#F505);		
+	
+	ffZos : FB_FastFault := (
+        i_xAutoReset := TRUE,
+        i_DevName := 'T1K1 ZOS',
+        i_Desc := 'ST1K1 ZOS is out of Single Point Operation safe range',
```
Comment (ZLLentz) [inline lcls-plc-rixs-optics/rixs_optics/POUs/PRG_SPO_PMPS.TcPOU:37]: Would it be helpful to the user to include the ranges in these descriptions?

**Pull request merged**